numpy/testing/utils.py

```diff
@@ -666,9 +666,11 @@ def assert_approx_equal(actual,desired,significant=7,err_msg='',verbose=True):
 
 
 def assert_array_compare(comparison, x, y, err_msg='', verbose=True,
-                         header='', precision=6, equal_nan=True):
+                         header='', precision=6, equal_nan=True,
+                         equal_inf=True):
     __tracebackhide__ = True  # Hide traceback for py.test
-    from numpy.core import array, isnan, isinf, any, all, inf
+    from numpy.core import array, isnan, isinf, any, all, inf, zeros_like
+    from numpy.core.numerictypes import bool_
```

Comment:
@charris do you know why we have the imports here always? Circular imports I guess? Could put it to the top of the file, but again, seems fine can stay for another time.